ecommerce/EarningController.php

```diff
@@ -0,0 +1,300 @@
+<?php
+/**
+ * Manage earnings
+ *
+ * @package Tutor\Ecommerce
+ * @author Themeum <support@themeum.com>
+ * @link https://themeum.com
+ * @since 3.0.0
+ */
+
+namespace Tutor\Ecommerce;
+
+use Tutor\Helpers\QueryHelper;
+use Tutor\Models\OrderModel;
+
+/**
+ * Manage earnings
+ */
+class EarningController {
+
+	/**
+	 * Error message for the invalid earning data
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var string
+	 */
+	const INVALID_DATA_MSG = 'Invalid earning data';
+
+	/**
+	 * EarningController instance
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var EarningController object
+	 */
+	public static $instance = null;
+
+	/**
+	 * Earning table name
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var string
+	 */
+	private $earning_table;
+
+	/**
+	 * Order id
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var int
+	 */
+	private $order_id;
+
+	/**
+	 * Keep earning data here
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	public $earning_data = array();
```

Comment:
Make this property as `private` to ensure only this class method `prepare_order_earnings` can modify this.

To get safe prepared data, make a getter method
```php
public function get_earning_data() {
   return $this->earning_data;
}
```

Later we can use
```php
if ( $earning_id ) {
    do_action( 'tutor_ecommerce_after_earning_stored', $earning_id, $earnings->get_earning_data() );
}
```

Reply:
`public $earning_data = array();` earning data need to keep public. `prepare_order_earnings` method is a special case. We may not need to prepare earning_data using order id always.